metrics/test_driver/process_manager/process_manager.dart

```diff
@@ -0,0 +1,56 @@
+import 'dart:io';
+
+import '../cli/common/logger/process_logger.dart';
+import '../cli/common/runner/process_runner.dart';
+
+/// Starts new processes and then manages them.
+class ProcessManager {
+  final List<Process> _startedProcesses = [];
+
+  /// Runs the process using the [runner], subscribes to logs, and saves
+  /// the process to be able to kill it.
+  Future<Process> run(
+    ProcessRunner runner, {
+    String workingDir,
+    String logFileName,
+  }) async {
+    final process = await runner.run(
+      workingDir: workingDir,
+    );
+
+    final runnerFinishedFuture = runner.started;
+
+    _startedProcesses.add(process);
+
+    process.stderrBroadcast.listen((error) => _processErrorHandler(process));
+
+    if (logFileName != null) {
+      ProcessLogger.startLogging(process, logFileName);
+    }
+
+    await runnerFinishedFuture;
+
+    return process;
+  }
+
+  /// Closes the application on process error.
+  Future<void> _processErrorHandler(Process process) async {
+    exitCode = await process.exitCode;
+    dispose();
+    exit(exitCode);
```

Comment:
I think we should better exit from the app in the FlutterWebDriver - as it's not obvious that it's happening here.